src/mesh/boundary_nonuniform.cxx.in.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+from jinja2 import Environment
+import stencils_sympy as sten
+
+header="""\
+#include <boundary_standard.hxx>
+#include <bout/constants.hxx>
+#include <boutexception.hxx>
+#include <derivs.hxx>
+#include <fft.hxx>
+#include <globals.hxx>
+#include <invert_laplace.hxx>
+#include <msg_stack.hxx>
+#include <output.hxx>
+#include <utils.hxx>
+
+#include "boundary_nonuniform.hxx"
+
+static void update_stagger_offsets(int& x_boundary_offset, int& y_boundary_offset, int& stagger, CELL_LOC loc){
```

Comment:
I could get used to the C++17 version, assuming I can do something like:
```
const auto [y_boundary_offset, stagger] = offset<CELL_YLOW>(bndry->by, stagger, loc);
```
I.e. update stagger.

The other version requires to much handling of extracting stagger, to so that the function is ~6 lines of code, and the calling side requires ~ 4.

In that case I would rather put it directly in the code, then call a function.
For the generated code, this isn't really an issue ...